### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by IamVanilla**
I finished unit tests for the library

### CODE REVIEW

File: .nyc_output/1cbd51b1-47e0-4420-aa9c-83e9ada0dfba.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (hubjac1) [inline .nyc_output/1cbd51b1-47e0-4420-aa9c-83e9ada0dfba.json]: remove this file

File: .nyc_output/517f1366-34dd-40db-8e02-35d0dbc4a478.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (hubjac1) [inline .nyc_output/517f1366-34dd-40db-8e02-35d0dbc4a478.json]: remove this file

File: package.json [hunk 3]
```diff
@@ -19,7 +19,8 @@
 		"build": "npm run tslint && tsc",
 		"tslint": "tslint src/*",
 		"test:debug": "karma start",
-		"test": "karma start  --single-run"
+		"test": "karma start  --single-run",
+		"tescov": "nyc mocha"
```
Comment (hubjac1) [inline package.json]: remove this command

File: package.json [hunk 4]
```diff
@@ -37,6 +38,7 @@
 		"karma-mocha": "^1.3.0",
 		"karma-typescript": "^3.0.13",
 		"mocha": "^5.2.0",
+		"nyc": "^13.3.0",
```
Comment (hubjac1) [inline package.json]: useless dependency

File: test/test-basic.ts [hunk 5]
```diff
@@ -128,11 +129,78 @@ describe('httpclient', () => {
 			assert.notEqual(aRequest.timeout, 30000)
```
Comment (hubjac1) [inline test/test-basic.ts]: I's better to test it the properties are equal to the new value

File: test/test-basic.ts [hunk 6]
```diff
@@ -128,11 +129,78 @@ describe('httpclient', () => {
...
+			packOfFilters[1] = new httpclient.InstalledFilter(new FourthStep() , '4')
+			packOfFilters[2] = new httpclient.InstalledFilter(new FifthStep() , '5')
+			let filterCollection = new httpclient.FilterCollection(packOfFilters)
+			httpClient.addFilter(new FirstStep(), '1')
+			httpClient.addFilter(new SecondStep(), '2')
+			httpClient.addFilter(filterCollection, '3,4,5')
+			httpClient.addFilter(new LastStep(), '6')
+			mainFilterChain = new FilterChainImpl((httpClient as any)._filters)
+			const theResponse = await mainFilterChain.doFilter(aRequest)
+			assert.equal((theResponse.request.body as Post).body, '(╯°□°）╯︵ ┻━┻')
```
Comment (hubjac1) [inline test/test-basic.ts:204]: 😄

File: test/test-unitaires-execute.ts [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+	describe('execute with a real API external', function() {
+		let postResponse: Object
+		let idOfCreatedObject: number
+		before(async function() {
+			this.timeout(5000)
+			aRequest = new httpclient.Request('http://dummy.restapiexample.com/api/v1/create')
+			aRequest.method = 'POST'
+			aRequest.body = {
+				'name': 'auDD',
+				'salary': '333',
```
Comment (hubjac1) [inline test/test-unitaires-execute.ts]: 333k

File: test/test-unitaires-execute.ts [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+			try {
+				aRequest = new httpclient.Request('i@#ccidently_mizspeld976theurl/')
+				await execute(aRequest)
+				assert.isTrue(false, 'previous line should throw an error')
+			} catch (error) {
+				const response: httpclient.Response<Object> = error
+				assert.equal(response.status, 404)
+			}
+		})
+		it('should also throw an error', async function() {
```
Comment (hubjac1) [inline test/test-unitaires-execute.ts]: could you be more descriptive?

**Comment by hubjac1** [comment]
Good job, I left a few comments.

File: src/execute.ts [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+							const header = parts.shift()!
+							headerMap[header] = parts.join(': ')
+						}
+					}
+				})
+			}
+			return headerMap
+		}
+
+		// This inernal method takes an XMLHttpRequest and will return a Response
```
Comment (antoinebaudoux) [inline src/execute.ts]: Typo "inernal"

**Review by antoinebaudoux — CHANGES_REQUESTED** [review]

**Review by antoinebaudoux — APPROVED** [review]

**Pull request merged**